docker/Dockerfile

```diff
@@ -92,16 +50,10 @@ RUN apt-get -y update \
       # misc
       make \
       git \
+      python${PYTHON_VERSION}-dev \
```

Comment:
we use uv's Python so this shouldn't be necessary

Reply:
Ah I see that you switched to pip from uv, I would like to keep it, it's pretty much a drop-in replacement but much faster and nicer